src/seedcase_sprout/core/sprout_checks/_check_column_types.py

```diff
@@ -0,0 +1,118 @@
+from typing import Callable, TypedDict
+
+import polars as pl
+
+from seedcase_sprout.core.get_nested_attr import get_nested_attr
+from seedcase_sprout.core.properties import (
+    FieldProperties,
+    FieldType,
+    ResourceProperties,
+)
+
+
+def _check_column_types(
+    data: pl.DataFrame, resource_properties: ResourceProperties
+) -> pl.DataFrame:
+    """Checks that column data types match the data types specified in the properties.
+
+    The resource properties specify a Frictionless data type for each column.
+    This function checks if the Polars data type of each column in the data is an
+    allowed realisation of the corresponding Frictionless data type.
+
+    Column names are expected to match the names specified in the resource properties.
+
+    Args:
+        data: The data frame to check.
+        resource_properties: The resource properties to check against.
+
+    Returns:
+        The data frame, if all column types are correct.
+
+    Raises:
+        ExceptionGroup: A group of `ValueError`s, one per incorrectly typed column.
+    """
+    all_column_props: list[FieldProperties] = get_nested_attr(
```

Comment:
```suggestion
    field_properties: list[FieldProperties] = get_nested_attr(
```

Or `column_properties`

Reply:
If you don't mind using "field" in the name, then I'll just go for `fields`